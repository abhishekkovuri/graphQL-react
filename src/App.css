.list {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh;
}
header {
  font-size: 36px;
  font-weight: 800;
}

.item {
  border: 1px solid #999;
  border-radius: 10px;
  margin: 10px;
  padding: 10px;
  width: 250px;
  text-align: center;
  font-size: 18px;
  font-weight: 800;
}