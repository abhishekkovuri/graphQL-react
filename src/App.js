import React from 'react';
import { useLazyQuery, useQuery } from '@apollo/client';
import { GET_COUNTRIES } from './graphql-query/getCountries.js';
import './App.css';

const App = () => {
	const { data } = useQuery(GET_COUNTRIES);
	const countries = data?.getCountries?.countries;

	const [lazyLoadBtn, { data: duplicateCountriesData }] =
		useLazyQuery(GET_COUNTRIES);
	const duplicateCountries = duplicateCountriesData?.getCountries?.countries;

	return (
		<section className="list">
			<header> Render Country List </header>
			<section>
				<h1>Original data</h1>
				{countries &&
					countries.map((country) => (
						<div className="item" key={country.countryName}>
							{country.countryName} - {country.capital}
						</div>
					))}
			</section>
			<button onClick={lazyLoadBtn}>Click me</button>
			<section>
				<h1>Duplicate data</h1>
				{duplicateCountries &&
					duplicateCountries.map((country) => (
						<div className="item" key={country.countryName}>
							{country.countryName} - {country.capital}
						</div>
					))}
			</section>
		</section>
	);
};

export default App;
