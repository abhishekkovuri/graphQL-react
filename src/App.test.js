import { getByText, render } from '@testing-library/react';
import { useQuery, useLazyQuery } from '@apollo/client';
import App from './App';

jest.mock('@apollo/client', () => {
	return {
		gql: jest.fn(),
		useQuery: jest.fn(),
		useLazyQuery: jest.fn(),
	};
});

describe('App test', () => {
	beforeEach(() => {
		useQuery.mockReturnValue({
			data: {
				getCountries: {
					countries: [
						{
							countryName: 'India',
							capital: 'New Delhi',
						},
					],
				},
			},
		});
		useLazyQuery.mockReturnValue([
			jest.fn(),
			{
				data: {
					getCountries: {
						countries: [
							{
								countryName: 'Japan',
								capital: 'Tokyo',
							},
						],
					},
				},
			},
		]);
	});

	test('should generate snapshots', () => {
		const { asFragment } = render(<App />);

		expect(asFragment()).toMatchSnapshot();
	});

	test('should display the ', () => {
		const { getByText, queryByText } = render(
			<App />
		);

		expect(getByText(/Render Country List/i)).toBeInTheDocument();
		expect(getByText(/India - New Delhi/i)).toBeInTheDocument();
		expect(queryByText(/Japan - Tokyo/i)).toBeInTheDocument();
	});
});
