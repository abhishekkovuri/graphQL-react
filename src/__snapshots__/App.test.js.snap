// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should generate snapshots 1`] = `
<DocumentFragment>
  <section
    class="list"
  >
    <header>
       Render Country List 
    </header>
    <section>
      <h1>
        Original data
      </h1>
      <div
        class="item"
      >
        India - New Delhi
      </div>
    </section>
    <button>
      Click me
    </button>
    <section>
      <h1>
        Duplicate data
      </h1>
      <div
        class="item"
      >
        Japan - Tokyo
      </div>
    </section>
  </section>
</DocumentFragment>
`;
